enhancements/installer/telemeter-metrics.md

```diff
@@ -0,0 +1,300 @@
+cluster-ingress-02-config.yml              etcd-ca-bundle-configmap.yaml    etcd-metric-signer-secret.yaml         kube-system-configmap-root-ca.yaml
+```
+
+The number of potential values is approximately 30.
+
+Either of these proposed metrics would be pushed only upon successful completion of the `create cluster` command.
+
+#### Metric Labels
+Collecting labels into a single metric sample and pushing once means that each sample represents a single installer invocation. When a sample is pushed to the aggregation pushgateway metrics with matching labels will be incremented:
+
+```
+cluster_installation_invocation{command="create", target="cluster",os="linux",platform="AWS",result="ClusterComplete",return="0",version="4.2",duration="25"} 100
+cluster_installation_invocation{command="create", target="cluster",os="linux",platform="GCP",result="APIFailure",return="1",version="4.3",duration="30"} 25
+cluster_installation_invocation{command="destroy", target="cluster",os="linux",platform="AWS",result="Success",return="1",version="4.2",duration="10"} 5
+```
+If the above example represents the state of our Prometheus database, it would indicate that there have been 100 successful cluster installations on AWS that took 25 or fewer minutes.
+
+The values for these labels may only become available at different times during an installation or command execution. Therefore the installer instrumentation must handle building labels by collecting key value pairs as they become available.
+
+#### Internal Installer Design
+A new metrics package would be created to handle the collection of metrics, building labels, and pushing to the aggregation pushgateway. 
+
+As required label values are gathered throughout many installer packages,  the `metrics` package would maintain state package-scoped variables (feedback please: package-scoped variables may be generally frowned upon but is this a valid use case?).
+
+##### API
+The package acts as a lightweight wrapper around the [Prometheus SDK][prom-godocs]. The `metrics` package would expose the following functions to be used by other packages within the installer:
+```
+// AddLabelValue adds a key/value pair to the label being built for metric.
+AddLabelValue(metric, key, value string)
+
+// Pushes the specified metric to the gateway.
+Push(metric string)
+
+// SetValue sets the value of the metric. The value would always be 1 for the present design. This function is included for completeness.
+SetValue(metric, value string)
+```
+
+The metric `cluster_installation_modification` would require a separate API function based on a choice of the design decisions listed above. In the second case, where each modified asset is sent in a separate sample, the API function would simply wrap the Prometheus SDK call to add to a metrics vector:
+
+```
+// AddToMetric adds a sample to a metrics vector.
+AddToMetric(metric string, labels map[string]string, value string)
+```
+
```

Comment:
internal implementation should also probably also include

- how to stop the metrics
- exactly when the metrics are pushed.

Reply:
```suggestion
// StopMetrics is used to stop the system from sending metrics to the server when the user prefers not to disclose information by giving a flag to the installer.
StopMetrics()
```

Reply:
I think we should limit this enhancement to the user interaction with the installer and the installer's interaction with Prometheus, i.e. the metrics design. @rna-afk will work on a separate enhancement solely focused on installer implementation/instrumentation. 

Regarding disabling metrics collection, in this doc we should discuss: is it a command-line flag, environment variable, install config setting, other? 

@abhinavdahiya let me know if that sounds acceptable to split this